### PR TITLE
Update model during RenderableModelEntityItem update

### PR DESCRIPTION
**Pull request by sethalves**
- partial fix for a bug where held model entities would jitter around when the avatar turns

### CODE REVIEW

**Comment by ZappoMan** [comment]
don't you want lines 220-260 as well?

**Comment by sethalves** [comment]
no, those refer to the scene and the render args

**Comment by hifi-gustavo** [comment]
Build console output for commit: 283ad7ffd938a858510f815bb4442e13e89ef0ff available at: https://gist.github.com/a69282abd58792706274<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6474.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6474.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: cfcff42004cd134224bbd46ff49dcf35c07a7b63 available at: https://gist.github.com/a69282abd58792706274<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6474.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6474.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5cd047da15cfdc65d4e2a23700a0f3ed99ebd5b0 available at: https://gist.github.com/a69282abd58792706274<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6474.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6474.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 09f14bc9a56bc7bb54c2c24a773058b9beed7ab0 available at: https://gist.github.com/a69282abd58792706274<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6474.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6474.dmg

**Comment by howard-stearns** [comment]
This is not rendering all models for me. (Mac PR build 3367)
It is as though I didn't have everything cached. However, I can switch to master and it renders fine, and then I switch back to the PR build and models don't show again, but not always the same ones

**Comment by hifi-gustavo** [comment]
Build console output for commit: c96a09976e16f987151b9002ab9143a1d5f765c1 available at: https://gist.github.com/a69282abd58792706274<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6474.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6474.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4ab2bfa85461d7e1fd7b298dfb08b8b10c1cbd2e available at: https://gist.github.com/a69282abd58792706274<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6474.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6474.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4b5fec4b008731a905e87ee669d0827a2af4d5f2 available at: https://gist.github.com/a69282abd58792706274<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6474.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6474.dmg

**Comment by sethalves** [comment]
@howard-stearns could you retest?  I attempted to clean up the if-maze a little.

File: libraries/entities-renderer/src/RenderableModelEntityItem.cpp [hunk 1]
```diff
@@ -360,7 +306,35 @@ Model* RenderableModelEntityItem::getModel(EntityTreeRenderer* renderer) {
...
+    }
+
+    if (_myRenderer && (!_model || _needsModelReload)) {
+        return true;
+    }
+
+    if (_model) {
+        if (hasAnimation() || jointsMapped()) {
+            return true;
+        }
```
Comment (AndrewMeadows) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: The above **hasAnimation()** is **true** when the animation URL property is not empty (and we return **true** for that case).

Meanwhile the below **isAnimatingSomething()** is **true** not only when the animation URL is not empty but the animation has a specified FPS and is set playing (and we return **true** for that case... but it will never be true in this context since if it were we would already have returned **true** above.

In short, the **isAnimatingSomething()** below is not helpful.
